### PR TITLE
feat: Create keyboard listeners

### PR DESCRIPTION
**Pull request by mingard**
feat: Basic keyboard event lib

@eduardoboucas Still needs:
- [ ] OS variants
- [ ] Shortcut shortcuts e.g. `keyboard.DELETE`
- [x] `off` method
- [x] map all keys

### CODE REVIEW

File: dadi/frontend/lib/keyboard.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  constructor (pattern, keys) {
+    this.pattern = pattern
+    this.keys = keys
+    this.active = 0
+  }
+
+  next (keyCode) {
+    if (this.match(keyCode, this.keys[this.active])) {
+      if (this.active < this.keys.length -1) {
+        this.active ++
```
Comment (eduardoboucas) [inline dadi/frontend/lib/keyboard.js]: I'd prefer `this.active++` (without the space)

File: dadi/frontend/lib/keyboard.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    return this.shortcuts.find(shortcut => {
+      return shortcut.next(keyCode)
+    })
+  }
+
+  on (pattern) {
+    let keys = this.find(pattern.toLowerCase().split('+'))
+    let shortcut = new Pattern(pattern, keys)
+    this.shortcuts.push(shortcut)
+    return shortcut
```
Comment (eduardoboucas) [inline dadi/frontend/lib/keyboard.js]: It'd be great to have more consistency around spacing (perhaps something we could add to the doc). I think having a blank line after the declarations bit of every block (so after the two `let`) and before the `return` would make things more readable. You've done that in some parts of the PR.

Comment (mingard) [inline dadi/frontend/lib/keyboard.js]: @eduardoboucas should be better now

File: dadi/frontend/lib/keyboard.js (hunk 1 above)
Comment (mingard) [inline dadi/frontend/lib/keyboard.js]: Done

**Pull request merged**